<?xml version="1.0"?>
<launch>  
<arg name="read_fast" default="false" />
<arg name="read_once" default="false" />
<arg name="repeat_delay" default="0.0" />
<arg name="Device_1" default="false" /><!-- device 1 Start or Pause -->
<arg name="Device_2" default="false" /><!-- device 2 Start or Pause -->
<arg name="Device_3" default="false" /><!-- device 3 Start or Pause -->
<arg name="Device_4" default="false" /><!-- device 4 Start or Pause -->
<param name="save_xyz"  value="false"/>
<param name="OutExport_path"  value=""/>
<param name="device_num"  value="true"/>
<param name="model" value="C-Fans-128" />

<!-- device 1 : ip_192.168.0.3  port_2014 -->
<group ns="ns1">
<param name="frame_id" value="world" />
<param name="RT" value="0.0,0.0,0.0,0.0,0.0,0.0" /><!-- RX,RY,RZ,TX,TY,TZ -->
<param name="model" value="C-Fans-128" />
<param name="DateGrade"  value="5.0_format"/>
<node pkg="rfans_driver" type="driver_node" name="rfans_driver" output="screen">
<param name="advertise_name" value="rfans_packets" />
<param name="control_name" value="rfans_control"/>
<param name="device_ip" value="192.168.0.3" />
<param name="device_port" value="2014" />
<param name="rps" value="10"/>
<param name="Is_Start" value="$(arg Device_1)"/>
<param name="readfile_path"  value=""/>
<param name="cfg_path"  value=""/>
<param name="use_double_echo" value="false"/>
<param name="use_gps" value="false"/>
<param name="read_fast" value="$(arg read_fast)" />
<param name="read_once" value="$(arg read_once)" />
<param name="repeat_delay" value="$(arg repeat_delay)" />
<param name ="cut_angle_range" value="360.0"/>
</node>
</group>

<!-- device 2 : ip_192.168.0.4  port_2016 -->
<group ns="ns2">
<param name="RT" value="0.0,0.0,0.0,0.0,0.0,0.0" /><!-- RX,RY,RZ,TX,TY,TZ -->
<param name="frame_id" value="world" />
<param name="model" value="C-Fans-128" />
<param name="DateGrade"  value="5.0_format"/>
<node pkg="rfans_driver" type="driver_node" name="rfans_driver" output="screen">
<param name="advertise_name" value="rfans_packets" />
<param name="control_name" value="rfans_control"/>
<param name="device_ip" value="192.168.0.4" />
<param name="device_port" value="2016" />
<param name="rps" value="10"/>
<param name="Is_Start" value="$(arg Device_2)"/>
<param name="readfile_path"  value=""/>
<param name="cfg_path"  value=""/>
<param name="use_double_echo" value="false"/>
<param name="use_gps" value="false"/>
<param name="read_fast" value="$(arg read_fast)" />
<param name="read_once" value="$(arg read_once)" />
<param name="repeat_delay" value="$(arg repeat_delay)" />
<param name ="cut_angle_range" value="360.0"/>
</node>
</group>

<!-- device 3 : ip_192.168.0.5  port_2017 -->
<group ns="ns3">
<param name="RT" value="0.0,0.0,0.0,0.0,0.0,0.0" /><!-- RX,RY,RZ,TX,TY,TZ -->
<param name="frame_id" value="world" />
<param name="model" value="C-Fans-128" />
<param name="DateGrade"  value="5.0_format"/>
<node pkg="rfans_driver" type="driver_node" name="rfans_driver" output="screen">
<param name="advertise_name" value="rfans_packets" />
<param name="control_name" value="rfans_control"/>
<param name="device_ip" value="192.168.0.5" />
<param name="device_port" value="2017" />
<param name="rps" value="10"/>
<param name="Is_Start" value="$(arg Device_3)"/>
<param name="readfile_path"  value=""/>
<param name="cfg_path"  value=""/>
<param name="use_double_echo" value="false"/>
<param name="use_gps" value="false"/>
<param name="read_fast" value="$(arg read_fast)" />
<param name="read_once" value="$(arg read_once)" />
<param name="repeat_delay" value="$(arg repeat_delay)" />
<param name ="cut_angle_range" value="360.0"/>
</node>
</group>

<!-- device 4: ip_192.168.0.6  port_2018 -->
<group ns="ns4">
<param name="RT" value="0.0,0.0,0.0,0.0,0.0,0.0" /><!-- RX,RY,RZ,TX,TY,TZ -->
<param name="frame_id" value="world" />
<param name="model" value="C-Fans-128" />
<param name="DateGrade"  value="5.0_format"/>
<node pkg="rfans_driver" type="driver_node" name="rfans_driver" output="screen">
<param name="advertise_name" value="rfans_packets" />
<param name="control_name" value="rfans_control"/>
<param name="device_ip" value="192.168.0.6" />
<param name="device_port" value="2018" />
<param name="rps" value="10"/>
<param name="Is_Start" value="$(arg Device_4)"/>
<param name="readfile_path"  value=""/>
<param name="cfg_path"  value=""/>
<param name="use_double_echo" value="false"/><!-- Whether to use use_double_echo -->
<param name="use_gps" value="false"/>
<param name="read_fast" value="$(arg read_fast)" />
<param name="read_once" value="$(arg read_once)" />
<param name="repeat_delay" value="$(arg repeat_delay)" />
<param name ="cut_angle_range" value="360.0"/>
</node>
</group>


<node pkg="rfans_driver" type="u_coordinate" name="u_coordinate" output="screen">
</node>

<!--<node pkg="rviz" type="rviz" name="rviz"
 args=" -d $(find rfans_driver)/multi_Rviz_cfg.rviz"  output="screen"/>-->

</launch>
